### PR TITLE
 FC-489: :speech_balloon: Enhance developer documentation

### PR DESCRIPTION
**Pull request by agalagade**

### CODE REVIEW

File: pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md [hunk 1]
```diff
@@ -256,6 +256,8 @@ Once you've reached the editor, it is time to develop the actual function, wheth
 
 #### Function Layout and Framework
 
+Our runtime is built using NodeJS LTS (Long Term Support), which is currently NodeJs 12. Therefore, you have access to all functionality offered by NodeJS.
```
Comment (LivePersonFaas) [inline pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md:259]: NodeJS

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md [hunk 2]
```diff
@@ -22,24 +22,24 @@ The amount of lambdas connected to a specific event is restricted and can be see
...
-|No Event|Used for lambdas which only can be invoked externally||
-|Third-Party Bots Custom Integration|Triggered when a bot, which is using the Third-Party Bots' Custom Integration receives a message|15|
-|Third-Party Bots Post Hook|Triggered when a bot, which is using the Third-Party Bots' Post Hooks receives a message. The function will be invoked after the vendor responded|15|
-|Third-Party Bots Pre Hook|Triggered when a bot, which is using the Third-Party Bots' Post Hooks receives a message. The function will be invoked before the vendor receives the payload|15|
-|Third-Party Bots Error Hook|Triggered when a bot, which is using the Third-Party Bots' Error Hooks is having an issue|15|
+|Event Name|Event Description|Event Source|Lambda Limit|
+|--- |--- |--- |--- |
+|Chat Post Survey E-Mail Transcript|Triggered after a chat conversation is finished and CSAT survey has been submitted. For more details on how to set up a function like this one, please see details [here](liveperson-functions-integrations-post-chat-survey-transcripts.html).|Chat Service (Consumer Post chat survey)|10|
```
Comment (LivePersonFaas) [inline pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md:27]: pls add a linebreak before `For more details...` - also for the other rows

**Review by LivePersonFaas — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md (hunk 2 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md]: updated

**Review by LivePersonFaas — APPROVED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md [hunk 3]
```diff
@@ -51,12 +51,28 @@ The Payload Tab shows the example payload, which highlights the structure of the
...
+Using the relevant button from the Settings tab, you are able to add new Environment variables. Clicking the button will open a dialogue window where you can provide a name and a value for your new variable.
 
 You can access the variable during runtime by using `process.env['YOUR_ENV']`. **Please be aware** that the value will be available in the form of a string. If the value is a number, you will have to parse it prior to using it. By using the trash icon, you can delete unwanted variables.
 
 **Developer discretion is advised**, please be sure not to use confidential data such as credentials or secrets in the environment variables as these are saved and available in text form to anyone with access to the account. Functions has a secret storage service that can be leveraged for this purpose.
 
+**Note**, they are certain reserved variables, that are used by the platform and therefore they are not available to the consumer. The UI will inform you if you attempt to use them. some of these variables are:
+`NODE_ENV`,
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md:61]: Actually on second thought would actually remove the list, no need to share these details.

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/toolbelt.md [hunk 4]
```diff
@@ -473,11 +473,11 @@ The method collects every message which contains a keyword in an array. It retri
 
 ### GDPR Util
 
-This method provides GDPR related functionality, such as deleting transcripts of a conversation. This Util works for **Messaging-Use Cases only**!
+This method provides GDPR related functionality, such as replacing files of a conversation. This Util works for **Messaging-Use Cases only**!
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/toolbelt.md:476]:
```suggestion
This method provides GDPR related functionality, such as replacing files in a conversation. This Util works for **Messaging-Use Cases only**!
```

**Review by Templum — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/toolbelt.md (hunk 4 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/toolbelt.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md (hunk 3 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md [hunk 5]
```diff
@@ -51,12 +51,14 @@ The Payload Tab shows the example payload, which highlights the structure of the
...
+Using the relevant button from the Settings tab, you are able to add new Environment variables. Clicking the button will open a dialogue window where you can provide a name and a value for your new variable.
 
 You can access the variable during runtime by using `process.env['YOUR_ENV']`. **Please be aware** that the value will be available in the form of a string. If the value is a number, you will have to parse it prior to using it. By using the trash icon, you can delete unwanted variables.
 
 **Developer discretion is advised**, please be sure not to use confidential data such as credentials or secrets in the environment variables as these are saved and available in text form to anyone with access to the account. Functions has a secret storage service that can be leveraged for this purpose.
 
+**Note**, they are certain reserved variables, that are used by the platform and therefore they are not available to the consumer. The UI will inform you if you attempt to use them.
```
Comment (chrbrt) [inline pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md:60]:
```suggestion
**Note**, there are certain reserved variables, that are used by the platform and therefore they are not available to the consumer. The UI will inform you if you attempt to use them. 
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/Integrations/messaging-conversations.md [hunk 6]
```diff
@@ -24,25 +24,14 @@ Along with the invocation, the function is sent a payload containing metadata re
...
 
-* TTR (Time to Respond)
-
-* Participants Change
-
-* Conversation Idle
+### Messaging events for Function Invocation
 
-* Conversation Routing
+Conversational Cloud messaging uses a series of "Conversation State Change Events" which get fired when specific actions or events occur within the conversation. You are able to use theses events to trigger functions within Functions.
```
Comment (chrbrt) [inline pages/Documents/DeveloperTools/LivePersonFunctions/Integrations/messaging-conversations.md:32]: Is there a link to the "Conversation State Change Events" for more specific information?

```suggestion
Conversational Cloud messaging uses a series of "Conversation State Change Events" which are fired when specific actions or events occur within the conversation. You are able to use theses events to trigger your functions.
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md [hunk 7]
```diff
@@ -214,7 +214,7 @@ The **Schedules** section allows to define a time-based execution of lambdas via
 The **Logs** section is for analysing your functions' past invocations by searching through the logs you defined. [More information on logging your function can be found here](function-as-a-service-developing-with-faas-overview.html).
 
 
-The **Settings** section provides access to the available settings of Functions. Currently, users are able to whitelist domains and maintain secrets (i.e. OAuth tokens).
+The **Settings** section provides access to the available settings of Functions. Currently, users are able to whitelist domains, email reporting and maintain secrets (i.e. OAuth tokens).
```
Comment (chrbrt) [inline pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md:217]:
```suggestion
The **Settings** section provides access to the available settings of Functions. Currently, users are able to whitelist domains, set up email reporting and maintain secrets (e.g. OAuth tokens).
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md [hunk 8]
```diff
@@ -276,6 +276,8 @@ function callback(error, result){}
 
 If during the runtime of your application no error has occurred, you can provide a **null** value as error. After a **30** second timeout, your function will be killed immediately, regardless of its error state.
```
Comment (chrbrt) [inline pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md:277]:
```suggestion
If during the runtime of your application no error has occurred, you can provide a **null** value as error. After a **30** second timeout, your function's execution will be terminated immediately, regardless of its error state.
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md [hunk 9]
```diff
@@ -22,24 +22,24 @@ The amount of lambdas connected to a specific event is restricted and can be see
...
+|Third-Party Bots Post Hook|Triggered when a bot, which is using the Third-Party Bots' Post Hooks receives a message. The function will be invoked after the vendor responded.<br/> For more details on how to set up a function like this one, please see details [here](liveperson-functions-integrations-conversational-ai.html).|Bot Connector|15|
+|Third-Party Bots Pre Hook|Triggered when a bot, which is using the Third-Party Bots' Post Hooks receives a message. The function will be invoked before the vendor receives the payload.<br/> For more details on how to set up a function like this one, please see details [here](liveperson-functions-integrations-conversational-ai.html).|Bot Connector|15|
+|Third-Party Bots Error Hook|Triggered when a bot, which is using the Third-Party Bots' Error Hooks is having an issue.<br/> For more details on how to set up a function like this one, please see details [here](liveperson-functions-integrations-conversational-ai.html).|Bot Connector|15|
```
Comment (chrbrt) [inline pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md:42]: Pease change the following events sources from -> to
Controller bot >> Conversational Exchange Service (Messaging Auto Messages)
Conversational Cloud (chat survey) >> Chat Service (Consumer Post chat survey)
Conversational Cloud (conv. commands) >> Agent Workspace (NAW only)

**Review by chrbrt — CHANGES_REQUESTED** [review]
Here are some changes

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md (hunk 9 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/events-templates.md:42]: The events sources are already set properly.

File: pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md (hunk 8 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md (hunk 7 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/getting-started.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/Integrations/messaging-conversations.md (hunk 6 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/Integrations/messaging-conversations.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md (hunk 5 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/DevelopingwithFaaS/overview.md]: updated

**Pull request merged**